components/builder-jobsrv/src/server/log.rs

```diff
@@ -0,0 +1,44 @@
+use error::Result;
+use std::fs;
+use std::path::PathBuf;
+
+/// Encapsulates the local filesystem directory in which in-process
+/// build job logs will be collected prior to being sent to long-term
+/// storage.
+#[derive(Clone, Debug)]
+pub struct LogDirectory {
+    path: PathBuf,
+}
+
+impl LogDirectory {
+    /// Create a new `LogDirectory` wrapping `path`.
+    pub fn new(path: &str) -> Self {
+        LogDirectory { path: PathBuf::from(path) }
+    }
+
+    /// Ensures that the specified log directory can be used.
+    ///
+    /// # Panics
+    /// * when the path does not exist
+    /// * when the path is not a directory
+    /// * when the path is not writable
+    pub fn validate(&self) -> Result<()> {
```

Comment:
This function returns a Result but all of the error cases are a panic. You can have it just simply return `()` if you did intend to use panics, but we typically recommend only using those if the developer absolutely has made a mistake